src/v/storage/batch_cache.cc

```diff
@@ -361,6 +364,30 @@ void batch_cache_index::truncate(model::offset offset) {
     }
 }
 
+void batch_cache::background_reclaimer::start() {
+    _reclaim_loop = ss::with_scheduling_group(
+      _sg, [this] { return reclaim_loop(); });
+}
+ss::future<> batch_cache::background_reclaimer::stop() {
+    _stopped = true;
+    _change.signal();
```

Comment:
an alternative to a condition variable that doesn't have the lingering question about if the predicates (although digging deep into seastar it does look ok--I'm not sure how much that is guaranteed at the API level--this is a problem that threaded condition variables have too..) are evaluated synchronously with waiting is the following:

```c++
ss::sempahore sem{0};
sem.signal(); // wakeup (could be low mem)
sem.signal(); // wakeup (could be stop)
...
for (;;) {
   auto units = std::max(_sem.current(), size_t(1));
   sem.wait(units);
   ...
}
```